### PR TITLE
Added the Test cases on URLHelperTests

### PR DESCRIPTION
**Pull request by doomerxe**
Added the test cases of

1. empty url array
2. url array with invalid urls

on the URLHelperTests

Fixes: #7627

Signed-off-by: Jiahan Xi <doomerXe@gmail.com>

### CODE REVIEW

File: test/functional/cmdLineTests/shareClassTests/URLHelperTests/APITests/NullPointerAndEmptyArrayTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	public void run() throws Throwable {
+		
+		URL[] urlsempty = new URL[0];
+		URL[] urlsnull = new URL[1];
+		urlsnull[0] = null;
+		
+		CustomURLClassLoader loaderempty = new CustomURLClassLoader(urlsempty);
+		CustomURLClassLoader loadernull = new CustomURLClassLoader(urlsnull);
+
+		System.out.println("\nTEST PASSED");
```
Comment (hangshao0) [inline test/functional/cmdLineTests/shareClassTests/URLHelperTests/APITests/NullPointerAndEmptyArrayTest.java]: You can add the test name in the message.

File: test/functional/cmdLineTests/shareClassTests/URLHelperTests/URLHelperTests.xml [hunk 2]
```diff
@@ -99,6 +99,28 @@
...
+		<output type="failure" caseSensitive="yes" regex="no">Exception:</output>
+	</test>
+	
+	<test id="APITests.NullPointerAndEmptyArrayTest" timeout="600" runPath=".">
+		<command>$JAVA_EXE$ $currentMode$ $BOOTCLASSPATH$ -Xtrace:print={j9jcl.162,j9jcl.163,j9jcl.486} APITests.NullPointerAndEmptyArrayTest </command>
+		<output type="success" caseSensitive="yes" regex="no">TEST PASSED</output>
+		<output type="required" caseSensitive="yes" regex="no">Exiting because URL count is 0</output>
+		<output type="required" caseSensitive="yes" regex="no" showMatch="yes">Entering with url 0000000000000000</output>
+		<output type="required" caseSensitive="yes" regex="no">Exiting with 0 as CallObjectMethod failed for URLgetPathID</output>
+		<output type="failure" caseSensitive="yes" regex="no">Error:</output>
```
Comment (hangshao0) [inline test/functional/cmdLineTests/shareClassTests/URLHelperTests/URLHelperTests.xml]: Please add another failure condition "ASSERTION FAILED"

**Comment by doomerxe** [comment]
Hi @hangshao0 , the changes have been made, could you please take a look? Thank you!

Also, the tests passed on x86-64_linux (8 & 11).

**Comment by pshipton** [comment]
jenkins test sanity plinux jdk8,jdk11

**Comment by pshipton** [comment]
This test is failing in some cases, reverting while we figure it out.

**Comment by doomerxe** [comment]
> This test is failing on 32-bit, reverting while we figure it out.
> `>> Required condition was not found: [Output match: Entering with url 0000000000000000]`
> 
> Actual output `Entering with url 00000000`

Thank you! I will fix the test and make another PR.

**Pull request merged**